tests-lit/tests/junk_detection/04_junk_detection_using_bitcode_compilation_flags_via_record_command_line_flag/sample.cpp

```diff
@@ -0,0 +1,26 @@
+#ifndef FLAG
+#error "FLAG is not defined"
+#endif
+
+int sum(int a, int b) {
+  return a + b;
+}
+
+int main() {
+  return sum(-2, 2);
+}
+
+/**
+; RUN: cd / && %CLANG_EXEC -fembed-bitcode -g -DFLAG=1 %s -o %s.exe
+; RUN: cd %CURRENT_DIR
+; RUN: (unset TERM; %MULL_EXEC -test-framework CustomTest -mutators=all -reporters=IDE -ide-reporter-show-killed %s.exe 2>&1; test $? = 0) | %FILECHECK_EXEC %s --strict-whitespace --match-full-lines --check-prefix=WITHOUT-RECORD-COMMAND-LINE
+; WITHOUT-RECORD-COMMAND-LINE-NOT:Found compilation flags in the input bitcode
+; WITHOUT-RECORD-COMMAND-LINE:{{^.*}}sample.cpp:5:13: warning: Survived: Remove Void Call: removed llvm.dbg.declare [remove_void_function_mutator]{{$}}
+
+; RUN: cd / && %CLANG_EXEC -fembed-bitcode -g -DFLAG=1 -grecord-command-line %s -o %s.exe
```

Comment:
Looks like the `-grecord-command-line` flag is only supported starting from LLVM 8. We need to find a way to run this integration test on LLVM >= 8.

Reply:
I'd suggest taking a look at `SourceLocation` class.